admin/sosreport

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash -eu
+
+#
+# CLI: Create system state report
+#
+
+# elevate self privileges
+[[ ${EUID} -ne 0 ]] && exec sudo $(realpath ${BASH_SOURCE[0]}) $@
+# prevent modification of executable path
+PATH="/bin:/sbin:/usr/bin:/usr/sbin"
+
+# parse command line options
+while [[ $# -gt 0 ]]; do
+  case "$1" in
+    -h | --help)
+      echo "Create system state report."
+      echo "Use: $0 [OPTION...]"
+      echo "  -h, --help    Print this help and exit"
+      exit 0
+      ;;
+    *)
+      echo "Invalid argument: $1" >&2
+      exit 1
+      ;;
+  esac
+  shift
+done
+
+dreport
```

Comment:
I think that it would be better to use the hostlogger D-Bus interface for creating archive.
It makes also available this archive in the WebUI.
As an example you can look at [our old implementation for Vesnin ](https://b.yadro.com/projects/SRV/repos/klish/browse/vesnin/scripts/logs/create.sh)

Reply:
Do you mean phosphor-debug-collector D-Bus? It actually does the same - calls dreport file.
But we lose execution ouput of dreport iself. I don't like the idea to hide stdout/stderr for console applications. And what if there is some problems with D-Bus or debug-collector is stopped working?

Reply:
> Do you mean phosphor-debug-collector D-Bus? It actually does the same - calls dreport file.

Yes, sure. I've mistyped.

> But we lose execution ouput of dreport iself. I don't like the idea to hide stdout/stderr for console applications. And what if there is some problems with D-Bus or debug-collector is stopped working?

The general goal of my proposal is keeping only one way for this action. 
For showing the output we could call something like `journalctl -u xyz.openbmc_project.Dump.Manager`. D-Bus might be monitored for example through `busctl monitor`. 
And I agree, it does not look as an easy solution. So I don't  insist, I just propose to think about.

Reply:
As I [said](https://github.com/YADRO-KNS/obmc-yadro-cli/pull/1#discussion_r416575679) I don't like this idea, but I have implemented both options.
BTW, I got this today: `May 06 17:44:01 phosphor-dump-manager[1500]: Error occurred during the sd_event_add_child call`, so I wouldn't recommend to call dreport over D-Bus in production.

Reply:
I still think we need a single control path. Errors in that path are a different matter. They must be fixed instead of routed around.

Reply:
System administrator can't fix such errors on production server.
Do you mean I should remove direct dreport invoke, leave D-Bus call only?

Reply:
No, the way you did is ok. First trying the right way and routing around if it didn't work is fine with me.